packages/dx-react-scheduler-material-ui/src/templates/appointment-form/recurrence/layouts/daily.jsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+import classNames from 'classnames';
+import {
+  NUMBER_EDITOR,
+  getRecurrenceOptions,
+  changeRecurrenceOptions,
+} from '@devexpress/dx-scheduler-core';
+
+const styles = ({ spacing }) => ({
+  label: {
+    width: '6.5em',
+  },
+  labelWithMargin: {
+    marginLeft: '1em',
+    width: 'calc((100% - 7.5em) * 4 / 7)',
+  },
+  textEditor: {
+    width: 'calc((100% - 7.5em) * 3 / 7)',
+    maxWidth: '8em',
+  },
+  grid: {
+    marginTop: spacing(1.75),
+  },
+});
+
+const DailyBase = ({
+  weeklyRecurrenceSelectorComponent,
+  radioGroupComponent,
+  textEditorComponent: TextEditor,
+  labelComponent: Label,
+  classes,
+  getMessage,
+  readOnly,
+  onFieldChange,
+  appointmentData,
+  selectComponent,
+  className,
+  formatDate,
+  ...restProps
+}) => {
+  const { rRule } = appointmentData;
+  const recurrenceOptions = React.useMemo(() => getRecurrenceOptions(rRule) || {}, [rRule]);
+
+  const changeRecurrenceInterval = React.useCallback(interval => interval > 0 && onFieldChange({
+    rRule: changeRecurrenceOptions({ ...recurrenceOptions, interval }),
+  }), [recurrenceOptions, onFieldChange]);
+  return (
+    <Grid
```

Comment:
Seems that the same whole Grid is used for all layouts so it can be extracted to a component

Reply:
I meant Grid's children too - only label text is different